staging/src/k8s.io/apiserver/pkg/server/openapi/openapi.go

```diff
@@ -17,35 +17,65 @@ limitations under the License.
 package openapi
 
 import (
+	"bytes"
+	"compress/gzip"
+	"crypto/sha512"
+	"encoding/json"
 	"fmt"
+	"gopkg.in/yaml.v2"
+	"mime"
 	"net/http"
 	"reflect"
 	"strings"
+	"time"
 
 	"github.com/emicklei/go-restful"
 	"github.com/go-openapi/spec"
+	"github.com/golang/protobuf/proto"
+	"github.com/googleapis/gnostic/OpenAPIv2"
+	"github.com/googleapis/gnostic/compiler"
 
 	"k8s.io/apimachinery/pkg/openapi"
-	"k8s.io/apimachinery/pkg/util/json"
 	genericmux "k8s.io/apiserver/pkg/server/mux"
 	"k8s.io/apiserver/pkg/util/trie"
 )
 
 const (
 	OpenAPIVersion  = "2.0"
 	extensionPrefix = "x-kubernetes-"
+
+	JSON_EXT = ".json"
+
+	MIME_JSON  = "application/json"
+	MIME_PB    = "application/com.github.googleapis.gnostic.OpenAPIv2+protobuf"
+	MIME_PB_GZ = "application/x-gzip"
```

Comment:
This is only zipped version of pb file. I am not sure how MIME suppose to work but I didn't see any example of zip file having unzipped content defined in MIME.